src/Akeneo/Pim/Structure/Bundle/Resources/public/js/attribute/form/useTranslateWithComponents.tsx

```diff
@@ -0,0 +1,81 @@
+import React, {ReactElement} from 'react';
+import {useTranslate} from '@akeneo-pim-community/shared';
+
+type ComponentFunction = (innerText: string) => ReactElement<any, any>;
+type ComponentsOrStrings = (string | ReactElement<any, any>)[];
+
+const splitPreviousElements: (
+  previousElements: ComponentsOrStrings,
+  componentName: string,
+  componentFunction: ComponentFunction
+) => ComponentsOrStrings = (previousElements, key, value) => {
```

Comment:
In order to do not rename the parameters because here it complexify the code
```suggestion
const splitPreviousElements: (
  previousElements: ComponentsOrStrings,
  componentName: string,
  componentFunction: ComponentFunction
): ComponentsOrStrings => {
```

Reply:
replaceComponentPlaceholder is maybe a better name